core/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala

```diff
@@ -94,12 +91,14 @@ private[spark] abstract class YarnSchedulerBackend(
   /**
    * An actor that communicates with the ApplicationMaster.
    */
-  private class YarnSchedulerActor extends Actor {
+  protected class YarnSchedulerActor(isDriver: Boolean)  extends Actor {
```

Comment:
this should be `isClusterMode` instead. I don't understand what `isDriver` means